src/pages/patientView/timeline2/VAFChartWrapper.tsx

```diff
@@ -42,99 +96,495 @@ const VAFPointConnector: React.FunctionComponent<{
+                    sampleId: dataPoint.sampleId,
+                    mutation: dataPoint.mutation,
+                    mutationStatus: dataPoint.mutationStatus,
+                });
+            });
+        });
+        return scaledData;
+    }
+
+    @computed get mutationToDataPoints() {
+        const map = new ComplexKeyMap<IPoint[]>();
+        for (const lineData of this.lineData) {
+            map.set(
+                {
+                    hugoGeneSymbol: lineData[0].mutation.gene.hugoGeneSymbol,
+                    proteinChange: lineData[0].mutation.proteinChange,
+                },
+                lineData
+            );
+        }
+        return map;
+    }
+
+    @computed get xposition() {
+        return this.props.store.getXPositionPerSample();
+    }
+
+    @computed get sampleIdOrder() {
+        return stringListToIndexSet(
+            this.props.sampleManager.getSampleIdsInOrder()
+        );
+    }
+
+    @computed get sampleGroupByColors() {
+        let groupByColors: { [s: string]: string } = {};
+        const uniqueColorGetter = makeUniqueColorGetter();
+        const clinicalAttributeSamplesMap = this.props.sampleManager.getClinicalAttributeSampleList(
+            this.props.sampleManager.samples,
+            this.props.store.groupByOption!
+        );
+        clinicalAttributeSamplesMap.forEach(
+            (sampleList: string[], clinicalValue: any) => {
+                groupByColors[clinicalValue] = uniqueColorGetter();
+            }
+        );
+        return groupByColors;
+    }
+
+    @computed get sampleGroupByLabels() {
+        let groupByLabels: string[] = [];
+        const uniqueColorGetter = makeUniqueColorGetter();
+        const clinicalAttributeSamplesMap = this.props.sampleManager.getClinicalAttributeSampleList(
+            this.props.sampleManager.samples,
+            this.props.store.groupByOption!
+        );
+        clinicalAttributeSamplesMap.forEach(
+            (sampleList: string[], clinicalValue: any) => {
+                groupByLabels.push(clinicalValue);
+            }
+        );
+        return groupByLabels;
+    }
+
+    @computed get sampleGroupByValue() {
+        let groupByValue: { [s: string]: string } = {};
+        if (
+            !(
+                this.props.store.groupByOption == null ||
+                this.props.store.groupByOption === 'None'
+            )
+        ) {
+            this.props.sampleManager.samples.forEach((sample, i) => {
+                groupByValue[
+                    sample.id
+                ] = SampleManager!.getClinicalAttributeInSample(
+                    sample,
+                    this.props.store.groupByOption!
+                )!.value;
+            });
+        }
+        return groupByValue;
+    }
+
+    @computed get sampleYPosition() {
+        // compute sample y position inside the footer
+        let ystart = 0;
+        let yposition: { [s: string]: number } = {};
```

Comment:
small thing but typically we do camelcase, so `yStart`, `yPosition`, etc

Reply:
and lets really try to be more specific in the typing. so `{ [sampleId: string]: number }`, for example. Even though its equivalent in typescript, it makes it 10x easier to read

Reply:
fixed